sql/catalog/lease: TestTableCreationPushesTxnsInRecentPast failed

sql/catalog/lease.TestTableCreationPushesTxnsInRecentPast [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12378352?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12378352?buildTab=artifacts#/) on release-23.2 @ [f958982e76095025f75933507f1c30dde1f8990e](https://github.com/cockroachdb/cockroach/commits/f958982e76095025f75933507f1c30dde1f8990e):


```
=== RUN   TestTableCreationPushesTxnsInRecentPast
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/1e31bde6494190748168afe7a747530a/logTestTableCreationPushesTxnsInRecentPast3251944325
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:82: cluster virtualization disabled due to issue: #109385 (expected label: C-test-failure)
    lease_test.go:2063: pq: relation "t.kv" does not exist
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1e31bde6494190748168afe7a747530a/logTestTableCreationPushesTxnsInRecentPast3251944325
--- FAIL: TestTableCreationPushesTxnsInRecentPast (6.98s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTableCreationPushesTxnsInRecentPast.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32813

--- a/pkg/sql/catalog/lease/BUILD.bazel
+++ b/pkg/sql/catalog/lease/BUILD.bazel
@@ -89,6 +89,7 @@ go_test(
         "//pkg/jobs/jobspb",
         "//pkg/keys",
         "//pkg/kv",
+        "//pkg/kv/kvclient/kvcoord",
         "//pkg/kv/kvpb",
         "//pkg/kv/kvserver",
         "//pkg/kv/kvserver/kvserverbase",

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -28,6 +28,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
@@ -2025,6 +2026,15 @@ func TestTableCreationPushesTxnsInRecentPast(t *testing.T) {
 
 	tc := serverutils.StartCluster(t, 3, base.TestClusterArgs{
 		ServerArgs: base.TestServerArgs{
+			Knobs: base.TestingKnobs{
+				KVClient: &kvcoord.ClientTestingKnobs{
+					// Intentionally add latency so that the uncertainty
+					// limit is higher to reduce the risk of this test flaking.
+					LatencyFunc: func(id roachpb.NodeID) (time.Duration, bool) {
+						return time.Millisecond * 100, true
+					},
+				},
+			},
 			DefaultTestTenant: base.TestDoesNotWorkWithSecondaryTenantsButWeDontKnowWhyYet(109385),
 		},
 		ReplicationMode: base.ReplicationManual,

